### PR TITLE
first try

### PR DESCRIPTION
**Pull request by unixfox**

### CODE REVIEW

File: src/core/endpoints/PlayerEndpoint.ts [hunk 1]
```diff
@@ -37,7 +37,10 @@ export function build(opts: PlayerEndpointOptions): IPlayerRequest {
     ...{
       client: opts.client,
       playlistId: opts.playlist_id,
-      params: opts.params
-    }
+      params: opts.params,
+      serviceIntegrityDimensions: {
+        poToken: session.po_token.toString(),
```
Comment (LuanRT) [inline src/core/endpoints/PlayerEndpoint.ts]: I saw you added `po_token` to the `PlayerEndpointOptions` type, so I think you meant `opts.po_token` here.

Comment (LuanRT) [reply src/core/endpoints/PlayerEndpoint.ts]: Other than that it looks good. But one last thing, you'll have to update the methods from the `Innertube` class that use this endpoint (`getInfo` & `getBasicInfo`).

File: src/core/Session.ts [hunk 2]
```diff
@@ -213,8 +217,9 @@ export default class Session extends EventEmitter {
   key: string;
   api_version: string;
   account_index: number;
+  po_token?: string;
 
-  constructor(context: Context, api_key: string, api_version: string, account_index: number, player?: Player, cookie?: string, fetch?: FetchFunction, cache?: ICache) {
+  constructor(context: Context, api_key: string, api_version: string, account_index: number, player?: Player, cookie?: string, fetch?: FetchFunction, cache?: ICache, po_token?: string) {
```
Comment (LuanRT) [inline src/core/Session.ts:222]: Ok there's another problem here actually. `po_token` is a parameter, but it's not being used anywhere. `Session#create` is not passing the token from `SessionOptions` when instantiating the class:

https://github.com/unixfox/YouTube.js/blob/6a131336840eec6f99235f8a829e31dd2b53ed5c/src/core/Session.ts#L271-L276

**Comment by unixfox** [comment]
thank you @LuanRT, I have opened a new PR here: https://github.com/LuanRT/YouTube.js/pull/708

**Pull request closed without merging**